src/Data/VarSet.hs

```diff
@@ -50,21 +47,19 @@ union = coerce Set.union
 intersection :: Set -> Set -> Set
 intersection = coerce Set.intersection
 
-singleton :: Var Resolved -> Set
-singleton (TgName _ x) = coerce (Set.singleton x)
-singleton _ = coerce Set.empty
+singleton :: CoVar -> Set
+singleton (CoVar x _ _) = coerce (Set.singleton x)
 
-delete :: Var Resolved -> Set -> Set
-delete (TgName _ x) set = coerce (Set.delete x (coerce set))
-delete _ set = set
+delete :: CoVar -> Set -> Set
+delete (CoVar x _ _) set = coerce (Set.delete x (coerce set))
 
 isEmpty :: Set -> Bool
 isEmpty = coerce Set.null
 
 class (Data a, Eq a, Ord a, Pretty a, Show a) => IsVar a where
```

Comment:
This could be in Core.Var.
Also drop the `Data a` constraint if you can.

Reply:
Yeah, I've been wanting to move it elsewhere for a while and wasn't sure where. This is as good a time as any!